tests/worker/test_workflow.py

```diff
@@ -7915,3 +7916,58 @@ async def test_external_activity_cancellation_details(
             assert err.details == temporalio.activity.ActivityCancellationDetails(
                 paused=True
             )
+
+
+@activity.defn
+async def short_activity_async():
+    delay = random.uniform(0.05, 0.15)  # 50~150ms delay
+    await asyncio.sleep(delay)
+    return 1
+
+
+@workflow.defn
+class QuickActivityWorkflow:
+    @workflow.run
+    async def run(self, total_seconds: float = 10.0):
+        workflow.logger.info("Duration: %f", total_seconds)
+        end = workflow.now() + timedelta(seconds=total_seconds)
+        while True:
+            workflow.logger.info("Stage 1")
+            res = await workflow.execute_activity(
+                short_activity_async, schedule_to_close_timeout=timedelta(seconds=10)
+            )
+            workflow.logger.info("Stage 2, %s", res)
+
+            if workflow.now() > end:
+                break
+
+
+async def test_quick_activity_swallows_cancellation(client: Client):
+    async with new_worker(
+        client,
+        QuickActivityWorkflow,
+        activities=[short_activity_async],
+        activity_executor=concurrent.futures.ThreadPoolExecutor(max_workers=1),
+    ) as worker:
+        temporalio.worker._workflow_instance._raise_on_cancelling_completed_activity_override = True
```

Comment:
Do you need to set this back to false at the end just in case `test_swallowed_activity_cancellation` runs after this?